shell_elvish.go

```diff
@@ -17,7 +17,7 @@ func (elvish) Hook() (string, error) {
 		if (> (count $m) 0) {
 			m = (explode $m)
 			keys $m | each [k]{
-				if (==s $k 'null') {
+				if (==s $m[$k] '') {
```

Comment:
I've changed my hook locally to  `if $m[$k]` which I think allows for an env var to be set to `''` which is what makes more sense I think. bash/elvish allow an env to be set to empty string and can tell the difference between set and empty vs not set. direnv also handles correctly.